src/sql/src/plan/query.rs

```diff
@@ -689,7 +695,7 @@ fn cast_relation<'a, I>(
     target_types: I,
 ) -> Result<HirRelationExpr, CastRelationError>
 where
-    I: IntoIterator<Item = &'a ScalarType>,
+    I: Iterator<Item = &'a ScalarType>,
```

Comment:
nit, but this looks like it should stay `IntoIterator` and the call to `into_iter()` should go where `target_types` is used: `target_types.into_iter().enumerate()`

Reply:
Ah! I had tried a few different permutations of this but not that one. I realize now that the type implements the trait, not that it itself is an iterator.